// @mantine/code-highlight/cjs/CodeHighlightTabs.cjs
'use client';
'use strict';

var React = require('react');
var hljs = require('highlight.js');
var cx = require('clsx');
var hooks = require('@mantine/hooks');
var core = require('@mantine/core');
var CopyIcon = require('./CopyIcon.cjs');
var FileIcon = require('./FileIcon.cjs');
var ExpandIcon = require('./ExpandIcon.cjs');
var CodeHighlight_module = require('./CodeHighlight.module.css.cjs');
var CodeHighlight_theme_module = require('./CodeHighlight.theme.module.css.cjs');

const classes = { ...CodeHighlight_module, root: cx(CodeHighlight_module.root, CodeHighlight_theme_module.theme) };
const defaultProps = {
  withHeader: true,
  copyLabel: "Copy code",
  copiedLabel: "Copied",
  maxCollapsedHeight: core.rem("8rem"),
  expandCodeLabel: "Expand code",
  collapseCodeLabel: "Collapse code"
};
const varsResolver = core.createVarsResolver((_, { maxCollapsedHeight }) => ({
  root: { "--ch-max-collapsed-height": core.rem(maxCollapsedHeight) }
}));
const CodeHighlightTabs = core.factory((_props, ref) => {
  const props = core.useProps("CodeHighlightTabs", defaultProps, _props);
  const {
    classNames,
    className,
    style,
    styles,
    unstyled,
    vars,
    children,
    code,
    defaultActiveTab,
    activeTab,
    onTabChange,
    withHeader,
    copiedLabel,
    copyLabel,
    getFileIcon,
    maxCollapsedHeight,
    expanded,
    defaultExpanded,
    onExpandedChange,
    expandCodeLabel,
    collapseCodeLabel,
    withExpandButton,
    ...others
  } = props;
  const getStyles = core.useStyles({
    name: "CodeHighlightTabs",
    props,
    classes,
    className,
    style,
    classNames,
    styles,
    unstyled,
    vars,
    varsResolver
  });
  const [value, setValue] = hooks.useUncontrolled({
    defaultValue: defaultActiveTab,
    value: activeTab,
    finalValue: 0,
    onChange: onTabChange
  });
  const [_expanded, setExpanded] = hooks.useUncontrolled({
    defaultValue: defaultExpanded,
    value: expanded,
    finalValue: true,
    onChange: onExpandedChange
  });
  const nodes = Array.isArray(code) ? code : [code];
  const currentCode = nodes[value];
  const highlighted = hljs.highlight(currentCode.code.trim(), {
    language: currentCode.language || "plaintext"
  }).value;
  const files = nodes.map((node, index) => /* @__PURE__ */ React.createElement(
    core.UnstyledButton,
    {
      ...getStyles("file"),
      key: node.fileName,
      mod: { active: index === value },
      onClick: () => setValue(index)
    },
    /* @__PURE__ */ React.createElement(
      FileIcon.FileIcon,
      {
        fileIcon: node.icon,
        getFileIcon,
        fileName: node.fileName,
        ...getStyles("fileIcon")
      }
    ),
    /* @__PURE__ */ React.createElement("span", null, node.fileName)
  ));
  return /* @__PURE__ */ React.createElement(core.Box, { ...getStyles("root"), mod: { collapsed: !_expanded }, ref, ...others, dir: "ltr" }, withHeader && /* @__PURE__ */ React.createElement("div", { ...getStyles("header") }, /* @__PURE__ */ React.createElement(core.ScrollArea, { type: "never", dir: "ltr", offsetScrollbars: false }, /* @__PURE__ */ React.createElement("div", { ...getStyles("files") }, files)), /* @__PURE__ */ React.createElement("div", { ...getStyles("controls") }, withExpandButton && /* @__PURE__ */ React.createElement(
    core.Tooltip,
    {
      label: _expanded ? collapseCodeLabel : expandCodeLabel,
      fz: "sm",
      position: "left"
    },
    /* @__PURE__ */ React.createElement(
      core.ActionIcon,
      {
        onClick: () => setExpanded(!_expanded),
        variant: "none",
        "aria-label": _expanded ? collapseCodeLabel : expandCodeLabel,
        ...getStyles("control")
      },
      /* @__PURE__ */ React.createElement(ExpandIcon.ExpandIcon, { expanded: _expanded })
    )
  ), /* @__PURE__ */ React.createElement(core.CopyButton, { value: currentCode.code.trim() }, ({ copied, copy }) => /* @__PURE__ */ React.createElement(core.Tooltip, { label: copied ? copiedLabel : copyLabel, fz: "sm", position: "left" }, /* @__PURE__ */ React.createElement(
    core.ActionIcon,
    {
      onClick: copy,
      variant: "none",
      ...getStyles("control"),
      "aria-label": copied ? copiedLabel : copyLabel
    },
    /* @__PURE__ */ React.createElement(CopyIcon.CopyIcon, { copied })
  ))))), /* @__PURE__ */ React.createElement(core.ScrollArea, { type: "auto", dir: "ltr", offsetScrollbars: false }, /* @__PURE__ */ React.createElement(core.Box, { ...getStyles("codeWrapper"), mod: { expanded: _expanded } }, /* @__PURE__ */ React.createElement("pre", { ...getStyles("pre") }, /* @__PURE__ */ React.createElement("code", { ...getStyles("code"), dangerouslySetInnerHTML: { __html: highlighted } })))), /* @__PURE__ */ React.createElement(
    core.UnstyledButton,
    {
      ...getStyles("showCodeButton"),
      mod: { hidden: _expanded },
      onClick: () => setExpanded(true)
    },
    expandCodeLabel
  ));
});
CodeHighlightTabs.displayName = "@mantine/core/CodeHighlightTabs";
CodeHighlightTabs.classes = classes;

exports.CodeHighlightTabs = CodeHighlightTabs;
