// @mantine/code-highlight/cjs/InlineCodeHighlight.cjs
'use client';
'use strict';

var React = require('react');
var cx = require('clsx');
var core = require('@mantine/core');
var hljs = require('highlight.js');
var CodeHighlight_module = require('./CodeHighlight.module.css.cjs');
var CodeHighlight_theme_module = require('./CodeHighlight.theme.module.css.cjs');

const classes = { ...CodeHighlight_module, code: cx(CodeHighlight_module.code, CodeHighlight_theme_module.theme) };
const defaultProps = {
  language: "tsx"
};
const InlineCodeHighlight = core.factory((_props, ref) => {
  const props = core.useProps("InlineCodeHighlight", defaultProps, _props);
  const { classNames, className, style, styles, unstyled, vars, code, language, ...others } = props;
  const getStyles = core.useStyles({
    name: "InlineCodeHighlight",
    props,
    classes,
    className,
    style,
    classNames,
    styles,
    unstyled,
    rootSelector: "code"
  });
  const highlighted = hljs.highlight(code.trim(), { language }).value;
  return /* @__PURE__ */ React.createElement(
    core.Box,
    {
      ...getStyles("code"),
      component: "code",
      ref,
      ...others,
      dangerouslySetInnerHTML: { __html: highlighted }
    }
  );
});
InlineCodeHighlight.displayName = "@mantine/core/InlineCodeHighlight";

exports.InlineCodeHighlight = InlineCodeHighlight;
